aptos-move/block-executor/src/executor.rs

```diff
@@ -122,11 +123,13 @@ where
     ) -> SchedulerTask {
         let _timer = TASK_EXECUTE_SECONDS.start_timer();
         let (idx_to_execute, incarnation) = version;
-        let txn = &signature_verified_block[idx_to_execute as usize];
+        let global_txn_idx = sharding_provider.global_idx_from_local(idx_to_execute);
+        sharding_provider.wait_for_remote_deps(idx_to_execute);
```

Comment:
@xosmig I asked for key-level dependencies but during implementation I realized I can let local txn wait for their dep txns and there's no need to touch the MVHashMap...